cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -375,11 +375,11 @@ private void executeFork(final ApplicationSpecification appSpec, WorkflowForkNod
           String branchInfo = retValue.getKey();
           WorkflowToken branchToken = retValue.getValue();
           ((BasicWorkflowToken) token).mergeToken(branchToken);
-          LOG.info("Execution of branch {} for fork {} completed", branchInfo, fork);
+          LOG.trace("Execution of branch {} for fork {} completed", branchInfo, fork);
         } catch (Throwable t) {
           Throwable rootCause = Throwables.getRootCause(t);
           if (rootCause instanceof ExecutionException) {
-            LOG.error("Exception occurred in the execution of the fork node {}.", fork, rootCause);
+            LOG.error("Exception occurred in the execution of the fork node {}.", fork);
```

Comment:
since `e.getCause()` is `Throwable` modified the suggestion slightly as 

```
 catch (InterruptedException e) {
          // Due to workflow abortion, so just break the loop
          break;
        } catch (ExecutionException e) {
          // Unwrap the cause
          Throwables.propagateIfPossible(e.getCause(), Exception.class);
          throw Throwables.propagate(e.getCause());
        }
```